wallet/wallet.go

```diff
@@ -3819,15 +3819,21 @@ func (w *Wallet) publishTransaction(tx *wire.MsgTx) (*chainhash.Hash, error) {
 		log.Warnf("Unable to remove invalid transaction %v: %v",
 			tx.TxHash(), dbErr)
 	} else {
-		// The spew output is pretty nice to directly see how many
-		// inputs/outputs of what type there are.
-		log.Infof("Removed invalid transaction: %v", spew.Sdump(tx))
+		log.Infof("Removed invalid transaction: %v", tx.TxHash())
 
-		// The serialized transaction is for logging only, don't fail on
-		// the error.
+		// The serialized transaction is for logging only, don't fail
+		// on the error.
 		var txRaw bytes.Buffer
 		_ = tx.Serialize(&txRaw)
-		log.Infof("Removed invalid transaction: %x", txRaw.Bytes())
+
+		// Optionally log the tx in debug when the size is manageable.
+		if txRaw.Len() < 1_000_000 {
+			log.Debugf("Removed invalid transaction: %v \n hex=%x",
+				spew.Sdump(tx), txRaw.Bytes())
```

Comment:
We can also use `newLogClosure` here. This way, spew is only called if the debug level would have us print the line. As is, it'll always be called to eval the args of the function.

Reply:
Done!